diff --git a/dock/process_info.go b/dock/process_info.go
index 83c61d58..e2970f3a 100644
--- a/dock/process_info.go
+++ b/dock/process_info.go
@@ -119,6 +119,16 @@ func verifyExe(exe, cwd, firstArg string) bool {
 		return false
 	}
 	logger.Debugf("firstArgPath: %q", firstArgPath)
+	if exe == firstArgPath {
+		return true
+	}
+	if strings.HasSuffix(exe, "-wrapped") {
+		exeBase := filepath.Base(exe)
+		if (len(exeBase) <= 9) {
+			return false
+		}
+		exe = exe[0:len(exe)-len(exeBase)] + exeBase[1:len(exeBase)-8]
+	}
 	return exe == firstArgPath
 }
 
