From 159ffabd16519c9c21155b367d98e0d09593e738 Mon Sep 17 00:00:00 2001
From: rewine <luhongxu@deepin.org>
Date: Tue, 31 Jan 2023 17:48:06 +0800
Subject: [PATCH] fix wrapped name for verifyExe

---
 dock/process_info.go | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/dock/process_info.go b/dock/process_info.go
index 83c61d58..cab8f7a5 100644
--- a/dock/process_info.go
+++ b/dock/process_info.go
@@ -119,6 +119,13 @@ func verifyExe(exe, cwd, firstArg string) bool {
 		return false
 	}
 	logger.Debugf("firstArgPath: %q", firstArgPath)
+	if exe == firstArgPath {
+		return true
+	}
+	if strings.HasSuffix(exe, "wrapped") {
+		exeBase := filepath.Base(exe)
+		exe = exe[0:len(exe)-len(exeBase)] + exeBase[1:len(exeBase)-8]
+	}
 	return exe == firstArgPath
 }
 
-- 
2.39.0

