From 2c77d679aa4e300a363b305bac251b229c921b7d Mon Sep 17 00:00:00 2001
From: rewine <lhongxu@outlook.com>
Date: Mon, 28 Nov 2022 20:21:06 +0800
Subject: [PATCH] patch_account_face_path_for_nix

---
 accounts/manager.go | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/accounts/manager.go b/accounts/manager.go
index 22c26ce7..663effb4 100644
--- a/accounts/manager.go
+++ b/accounts/manager.go
@@ -28,10 +28,10 @@ import (
 const (
 	actConfigDir       = "/var/lib/AccountsService"
 	userConfigDir      = actConfigDir + "/deepin/users"
-	userIconsDir       = actConfigDir + "/icons"
-	userCustomIconsDir = actConfigDir + "/icons/local"
+	userIconsDir       = @actConfigDir@ + "/icons"
+	userCustomIconsDir = actConfigDir + "/icons"
 
-	userIconGuest       = actConfigDir + "/icons/guest.png"
+	userIconGuest       = @actConfigDir@ + "/icons/guest.png"
 	actConfigFile       = actConfigDir + "/accounts.ini"
 	actConfigGroupGroup = "Accounts"
 	actConfigKeyGuest   = "AllowGuest"
-- 
2.38.0

