diff --git a/bin/dde-system-daemon/main.go b/bin/dde-system-daemon/main.go
index e39fa46b..95461df1 100644
--- a/bin/dde-system-daemon/main.go
+++ b/bin/dde-system-daemon/main.go
@@ -78,10 +78,6 @@ func main() {
 	// fix no PATH when was launched by dbus
 	if os.Getenv("PATH") == "" {
 		logger.Warning("No PATH found, manual special")
-		err = os.Setenv("PATH", "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin")
-		if err != nil {
-			logger.Warning(err)
-		}
 	}
 
 	// 系统级服务，无需设置LANG和LANGUAGE，保证翻译不受到影响
diff --git a/grub2/modify_manger.go b/grub2/modify_manger.go
index a811770b..30e9561e 100644
--- a/grub2/modify_manger.go
+++ b/grub2/modify_manger.go
@@ -21,7 +21,6 @@ const (
 )
 
 func init() {
-	_ = os.Setenv("PATH", "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin")
 }
 
 type modifyManager struct {
