diff --git a/accounts/user.go b/accounts/user.go
index 900033c6..f8827fb2 100644
--- a/accounts/user.go
+++ b/accounts/user.go
@@ -15,6 +15,7 @@ import (
 	"strings"
 	"sync"
 
+	"github.com/adrg/xdg"
 	dbus "github.com/godbus/dbus"
 	"github.com/linuxdeepin/dde-daemon/accounts/users"
 	authenticate "github.com/linuxdeepin/go-dbus-factory/com.deepin.daemon.authenticate"
@@ -649,7 +650,12 @@ func getUserSession(homeDir string) string {
 }
 
 func getSessionList() []string {
-	fileInfoList, err := ioutil.ReadDir("/usr/share/xsessions")
+	xsessionPath, err := xdg.SearchDataFile("xsessions")
+	if err != nil {
+		return nil;
+	}
+
+	fileInfoList, err := ioutil.ReadDir(xsessionPath)
 	if err != nil {
 		return nil
 	}
diff --git a/appearance/fsnotify.go b/appearance/fsnotify.go
index a409d0ba..ff674600 100644
--- a/appearance/fsnotify.go
+++ b/appearance/fsnotify.go
@@ -5,12 +5,15 @@
 package appearance
 
 import (
+	"errors"
+	"io/fs"
 	"os"
 	"path"
 	"path/filepath"
 	"strings"
 	"time"
 
+	"github.com/adrg/xdg"
 	"github.com/fsnotify/fsnotify"
 	"github.com/linuxdeepin/dde-daemon/appearance/background"
 	"github.com/linuxdeepin/dde-daemon/appearance/subthemes"
@@ -100,9 +103,14 @@ func (m *Manager) watchGtkDirs() {
 	gtkDirs = []string{
 		path.Join(home, ".local/share/themes"),
 		path.Join(home, ".themes"),
-		"/usr/local/share/themes",
-		"/usr/share/themes",
 	}
+	for _, dataPath := range xdg.DataDirs {
+                gtkPath := filepath.Join(dataPath, "themes");
+                if _, err := os.Stat(gtkPath); err != nil && errors.Is(err, fs.ErrNotExist) {
+                        continue
+                }
+                gtkDirs = append(gtkDirs, gtkPath);
+        }
 
 	m.watchDirs(gtkDirs)
 }
@@ -112,9 +120,14 @@ func (m *Manager) watchIconDirs() {
 	iconDirs = []string{
 		path.Join(home, ".local/share/icons"),
 		path.Join(home, ".icons"),
-		"/usr/local/share/icons",
-		"/usr/share/icons",
 	}
+	for _, dataPath := range xdg.DataDirs {
+                iconPath := filepath.Join(dataPath, "icons");
+                if _, err := os.Stat(iconPath); err != nil && errors.Is(err, fs.ErrNotExist) {
+                        continue
+                }
+                iconDirs = append(iconDirs, iconPath);
+        }
 
 	m.watchDirs(iconDirs)
 }
diff --git a/apps/utils.go b/apps/utils.go
index 8863d6c2..dd6f8e16 100644
--- a/apps/utils.go
+++ b/apps/utils.go
@@ -11,6 +11,7 @@ import (
 	"strconv"
 	"strings"
 	"syscall"
+	"github.com/adrg/xdg"
 )
 
 func intSliceContains(slice []int, a int) bool {
@@ -96,7 +97,7 @@ func removeDesktopExt(name string) string {
 }
 
 func getSystemDataDirs() []string {
-	return []string{"/usr/share", "/usr/local/share"}
+	return xdg.DataDirs
 }
 
 // get user home
