From d062bc282407537018db79209075f6583957d75a Mon Sep 17 00:00:00 2001
From: rewine <1758075541@qq.com>
Date: Mon, 11 Apr 2022 15:39:24 +0800
Subject: [PATCH] support use -DCMAKE_INSTALL_PREFIX flag

---
 CMakeLists.txt     | 17 ++++++++++++-----
 src/CMakeLists.txt |  5 +++--
 2 files changed, 15 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 77f9c88f..7c53c4b6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -68,12 +68,17 @@ endif()
 #模糊测试设置 -end
 
 #3.1 设置lib输出目录
-set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/lib)
+if(NOT DEFINED LIBRARY_OUTPUT_PATH)
+    set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/lib)
+endif()
 #3.2 设置可执行程序输出目录
-set(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR}/bin)
+if(NOT DEFINED EXECUTABLE_OUTPUT_PATH)
+    set(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR}/bin)
+endif()
 #导出头文件路径
-set(HFILES_OUTPUT_PATH ${CMAKE_BINARY_DIR}/include)
-
+if(NOT DEFINED HFILES_OUTPUT_PATH)
+    set(HFILES_OUTPUT_PATH ${CMAKE_BINARY_DIR}/include)
+endif()
 
 #Generate the qm file
 set(APP_RES_DIR "src")
@@ -91,7 +96,9 @@ if (USE_TEST)
 endif(USE_TEST)
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/src/config.h.in ${HFILES_OUTPUT_PATH}/config.h @ONLY)
 
-set(CMAKE_INSTALL_PREFIX /usr)
+if(NOT DEFINED CMAKE_INSTALL_PREFIX)
+    set(CMAKE_INSTALL_PREFIX /usr)
+endif()
 
 install(TARGETS cli7zplugin DESTINATION lib/deepin-compressor/plugins/)
 install(TARGETS clirarplugin DESTINATION lib/deepin-compressor/plugins/)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 153bda24..43c80f73 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -79,8 +79,9 @@ target_link_libraries(${EXE_NAME}
     compressor-interface
 )
 
-
-set(CMAKE_INSTALL_PREFIX /usr)
+if(NOT DEFINED CMAKE_INSTALL_PREFIX)
+    set(CMAKE_INSTALL_PREFIX /usr)
+endif()
 install(TARGETS deepin-compressor DESTINATION bin)
 install(FILES desktop/deepin-compressor.desktop DESTINATION share/applications)
 
-- 
2.35.1

