diff --git a/calendar-service/src/calendarprogramexitcontrol.cpp b/calendar-service/src/calendarprogramexitcontrol.cpp
index 34abac5c..16c24125 100644
--- a/calendar-service/src/calendarprogramexitcontrol.cpp
+++ b/calendar-service/src/calendarprogramexitcontrol.cpp
@@ -46,7 +46,6 @@ void CalendarProgramExitControl::reduce()
 void CalendarProgramExitControl::exit()
 {
 #ifdef NDEBUG
-    qApp->exit();
 #endif
 }
 
diff --git a/calendar-service/src/dbmanager/daccountmanagerdatabase.cpp b/calendar-service/src/dbmanager/daccountmanagerdatabase.cpp
index 8a504127..09d4464e 100644
--- a/calendar-service/src/dbmanager/daccountmanagerdatabase.cpp
+++ b/calendar-service/src/dbmanager/daccountmanagerdatabase.cpp
@@ -88,6 +88,7 @@ DAccount::Ptr DAccountManagerDataBase::getAccountByID(const QString &accountID)
 
 QString DAccountManagerDataBase::addAccountInfo(const DAccount::Ptr &accountInfo)
 {
+   qWarning() << "DBPath:" << getDBPath();
     SqliteQuery query(m_database);
     //生成唯一标识
     if (accountInfo->accountID().isEmpty()) {
diff --git a/calendar-service/src/synchronization/syncfilemanage.cpp b/calendar-service/src/synchronization/syncfilemanage.cpp
index 2a7d8b0b..c8fc6074 100644
--- a/calendar-service/src/synchronization/syncfilemanage.cpp
+++ b/calendar-service/src/synchronization/syncfilemanage.cpp
@@ -126,6 +126,7 @@ bool SyncFileManage::syncDataDelete(const QString &filepath)
 
 DAccount::Ptr SyncFileManage::getuserInfo()
 {
+    return nullptr;
     QVariantMap userInfoMap;
     //如果为社区版本则返回空
     if (Dtk::Core::DSysInfo::uosEditionType() == Dtk::Core::DSysInfo::UosCommunity) {
