diff --git a/calendar-service/src/dbusservice/dservicebase.cpp b/calendar-service/src/dbusservice/dservicebase.cpp
index ac182881..3324e0a2 100644
--- a/calendar-service/src/dbusservice/dservicebase.cpp
+++ b/calendar-service/src/dbusservice/dservicebase.cpp
@@ -40,6 +40,7 @@ QString DServiceBase::getClientName()
 
 bool DServiceBase::clientWhite(const int index)
 {
+    return true;
 //    DeepinAIAssista
 #ifdef CALENDAR_SERVICE_AUTO_EXIT
     //根据编号,获取不同到白名单
