From e715018edc050dce7a90ea17e81aa6118dd8e44b Mon Sep 17 00:00:00 2001
From: rewine <1758075541@qq.com>
Date: Wed, 27 Apr 2022 18:41:47 +0800
Subject: [PATCH] fix path

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bf2ddf7..80c8c35 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.7)
-
+include_directories(@qtbase_dev@)
 project(dde-control-center)
 
 find_package(DtkCMake REQUIRED)
-- 
2.35.3

