diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 45e8fb1..9fc5ff4 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -25,10 +25,11 @@ find_package(OpenMP REQUIRED)
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fopenmp")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fopenmp")
 
-pkg_check_modules(3rd_lib REQUIRED ncnn opencv_mobile)
+pkg_check_modules(3rd_lib REQUIRED ncnn)
+find_package(OpenCV 4 REQUIRED)
 
 target_include_directories(${PROJECT_NAME} PUBLIC ${3rd_lib_INCLUDE_DIRS})
-target_link_libraries(${PROJECT_NAME} ${3rd_lib_LIBRARIES} -lstdc++fs -ldl)
+target_link_libraries(${PROJECT_NAME} ${3rd_lib_LIBRARIES} opencv_core opencv_imgproc opencv_highgui -lstdc++fs -ldl)
 
 #install
 include(GNUInstallDirs)
