From 3ef764ce7db0437e05e185aec6971eef17dfb1c6 Mon Sep 17 00:00:00 2001
From: rewine <1758075541@qq.com>
Date: Thu, 31 Mar 2022 16:15:23 +0800
Subject: [PATCH 2/2] CMakeLists: use cmake install prefix

---
 src/CMakeLists.txt | 20 ++++++++++++--------
 1 file changed, 12 insertions(+), 8 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 9407d20..1144b7c 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -9,6 +9,10 @@ set(CMD_NAME deepin-camera)
 set(TARGET_NAME deepin-camera)
 #option (__mips__ "Use UNITTEST" ON)
 
+if(NOT DEFINED CMAKE_INSTALL_PREFIX)
+  set(CMAKE_INSTALL_PREFIX "/usr")
+endif()
+
 # cube文件目录，libvisualresult在打包时，会自动将cube文件安装到share目录下
 set(LUT_DIR "/usr/share/libimagevisualresult/filter_cube")
 add_definitions(-DLUT_DIR="${LUT_DIR}")
@@ -128,33 +132,33 @@ target_link_libraries(${TARGET_NAME} ${3rd_lib_LIBRARIES}
 qt5_use_modules(${TARGET_NAME} ${QtModule})
 
 #application
-install(TARGETS ${TARGET_NAME} DESTINATION /usr/bin)
+install(TARGETS ${TARGET_NAME} DESTINATION bin)
 
 #desk desktop
 install(FILES ${PROJECT_SOURCE_DIR}/deepin-camera.desktop
-    DESTINATION /usr/share/applications)
+    DESTINATION share/applications)
 
 #table desktop
 install(FILES ${PROJECT_SOURCE_DIR}/table/deepin-camera.desktop
-    DESTINATION /usr/share/due-shell/shortcuts)
+    DESTINATION share/due-shell/shortcuts)
 
 #icons
 install(FILES ${PROJECT_SOURCE_DIR}/assets/deepin-camera.svg
-    DESTINATION /usr/share/icons/hicolor/scalable/apps)
+    DESTINATION share/icons/hicolor/scalable/apps)
 
 #table_icons
-install(DIRECTORY ${PROJECT_SOURCE_DIR}/assets/panel/Icons DESTINATION /usr/share/due-shell/shortcuts)
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/assets/panel/Icons DESTINATION share/due-shell/shortcuts)
 
 #manual
-install(DIRECTORY ${PROJECT_SOURCE_DIR}/assets/deepin-camera DESTINATION /usr/share/deepin-manual/manual-assets/application/)
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/assets/deepin-camera DESTINATION share/deepin-manual/manual-assets/application/)
 
 #debus_service
-install(FILES ${PROJECT_SOURCE_DIR}/com.deepin.Camera.service DESTINATION /usr/share/dbus-1/services)
+install(FILES ${PROJECT_SOURCE_DIR}/com.deepin.Camera.service DESTINATION share/dbus-1/services)
 
 ##translations
 #install(FILES ${PROJECT_SOURCE_DIR}/translations/*.qm DESTINATION /usr/share/${PROJECT_SOURCE_DIR}/translations)
 
-set(APPSHAREDIR  /usr/share/${TARGET_NAME})
+set(APPSHAREDIR  share/${TARGET_NAME})
 
 #translations
 install(DIRECTORY ${PROJECT_SOURCE_DIR}/translations
-- 
2.35.1

